pkg/operation/botanist/addons.go

```diff
@@ -235,13 +237,18 @@ func (b *Botanist) generateCoreAddonsChart(ctx context.Context) (*chartrenderer.
 		return nil, err
 	}
 
+	monitoringEnabled := helper.IsMonitoringEnabled(b.Config)
```

Comment:
```suggestion
	monitoringEnabled := helper.IsMonitoringEnabled(b.Config) && b.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting
```
Just a suggestion to combine this and lines 241-243